asf_core_data/pipeline/preprocessing/feature_engineering.py

```diff
@@ -365,17 +370,13 @@ def get_postcode_coordinates(df):
     postcode_coordinates_df = coordinates.get_postcode_coordinates()
 
     # Reformat POSTCODE
-    df = data_cleaning.reformat_postcode(df)
-    postcode_coordinates_df = data_cleaning.reformat_postcode(postcode_coordinates_df)
-
-    postcode_coordinates_df["POSTCODE"] = (
-        postcode_coordinates_df["POSTCODE"].str.upper().str.replace(" ", "")
+    postcode_coordinates_df = data_cleaning.reformat_postcode(
+        postcode_coordinates_df, white_space="remove"
     )
+    df = data_cleaning.reformat_postcode(df, white_space="remove")
 
     # Merge with location data
-    df = pd.merge(df, postcode_coordinates_df, on=["POSTCODE"])
-
-    print(df.shape)
+    df = pd.merge(df, postcode_coordinates_df, on="POSTCODE")
```

Comment:
as default this does an inner merge - is that what you want? you'll lose a few lines from `df` if the postcode is missing.
even if it's what you want, might be helpful to make it explicit `how =  "inner"`

Reply:
Hm... I originally wanted to discard all samples without postcode, so I went for 'inner', but now I'm not so sure anymore. Maybe better to keep and filter at a later stage if we can't have samples without POSTCODE. Thanks for making think about that again!